NotificationsLab/app/src/main/java/com/ezequielc/notificationslab/MainActivity.java

```diff
@@ -0,0 +1,68 @@
+package com.ezequielc.notificationslab;
+
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.BitmapFactory;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.app.NotificationCompat;
+import android.widget.Toast;
+
+public class MainActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ConnectivityManager connectivityManager =
+                (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()){
+            showConnectionNotification();
+            Toast.makeText(this, "Connected to Network", Toast.LENGTH_SHORT).show();
+        } else {
+            showNoConnectionNotification();
+            Toast.makeText(this, "No Connection", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public void showConnectionNotification(){
+        Intent intent = new Intent(MainActivity.this, SecondActivity.class);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, (int) System.currentTimeMillis(), intent, 0);
+
+        NotificationCompat.BigPictureStyle bigPictureStyle = new NotificationCompat.BigPictureStyle();
+        bigPictureStyle.bigPicture(BitmapFactory.decodeResource(getResources(), R.drawable.wifi)).build();
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
+        builder.setSmallIcon(android.R.drawable.btn_star_big_on);
+        builder.setContentTitle("Notification Alert!");
+        builder.setAutoCancel(true); // dismiss the notification
+        builder.setStyle(bigPictureStyle);
+        builder.setContentIntent(pendingIntent);
+
+        NotificationManager notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+        notificationManager.notify(1, builder.build());
```

Comment:
Rather than hardcoding `1`, use a different constant for each notification. Then in the second activity, you can just call `cancel()` on the id for the "connected" notification, whether or not it is actually showing. No need to check for connectivity again from that activity.